service/src/main/java/uk/gov/hmcts/reform/fpl/service/cmo/ReviewCMOService.java

```diff
@@ -0,0 +1,119 @@
+package uk.gov.hmcts.reform.fpl.service.cmo;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.fpl.exceptions.CMONotFoundException;
+import uk.gov.hmcts.reform.fpl.exceptions.NoHearingBookingException;
+import uk.gov.hmcts.reform.fpl.model.CaseData;
+import uk.gov.hmcts.reform.fpl.model.HearingBooking;
+import uk.gov.hmcts.reform.fpl.model.ReviewDecision;
+import uk.gov.hmcts.reform.fpl.model.common.DocumentReference;
+import uk.gov.hmcts.reform.fpl.model.common.Element;
+import uk.gov.hmcts.reform.fpl.model.common.dynamic.DynamicList;
+import uk.gov.hmcts.reform.fpl.model.order.CaseManagementOrder;
+import uk.gov.hmcts.reform.fpl.service.time.Time;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+import static uk.gov.hmcts.reform.fpl.enums.CMOReviewOutcome.JUDGE_AMENDS_DRAFT;
+import static uk.gov.hmcts.reform.fpl.enums.CMOStatus.SEND_TO_JUDGE;
+import static uk.gov.hmcts.reform.fpl.model.order.CaseManagementOrder.sealFrom;
+import static uk.gov.hmcts.reform.fpl.utils.ElementUtils.asDynamicList;
+import static uk.gov.hmcts.reform.fpl.utils.ElementUtils.element;
+
+@Service
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+public class ReviewCMOService {
+
+    private final ObjectMapper mapper;
+    private final Time time;
+
+    public DynamicList buildDynamicList(List<Element<CaseManagementOrder>> cmos) {
+        return buildDynamicList(cmos, null);
+    }
+
+    public DynamicList buildDynamicList(List<Element<CaseManagementOrder>> cmos, UUID selected) {
+        return asDynamicList(cmos, selected, uk.gov.hmcts.reform.fpl.model.order.CaseManagementOrder::getHearing);
+    }
+
+    public Map<String, Object> handlePageDisplayLogic(List<Element<CaseManagementOrder>> cmos) {
+        Map<String, Object> data = new HashMap<>();
+        switch (cmos.size()) {
+            case 0:
+                data.put("numDraftCMOs", "NONE");
+                break;
+            case 1:
+                CaseManagementOrder cmo = cmos.get(0).getValue();
+                data.put("numDraftCMOs", "SINGLE");
+                data.put("reviewCMODecision",
+                    ReviewDecision.builder().hearing(cmo.getHearing()).document(cmo.getOrder()).build());
+                break;
+            default:
+                data.put("numDraftCMOs", "MULTI");
+                data.put("cmoToReviewList", buildDynamicList(cmos));
+                break;
+        }
+
+        return data;
+    }
+
+    public Element<CaseManagementOrder> getCMOToSeal(CaseData caseData, Element<CaseManagementOrder> cmo) {
```

Comment:
this method should accept only CaseData, then inside the method
Element<CaseManagementOrder> cmo = reviewCMOService.getSelectedCMO(caseData);

keep interface simple for client and coherent. Now you can pass CaseData and cmo which does not exists in case data, you never know what client pass to your interfcace